FIX _PLS transform or predict with copy=False

hzpc-joostk opened the pull request:
Fix for bug as described in https://github.com/scikit-learn/scikit-learn/issues/4122#issuecomment-78056759

Commit:
FIX _PLS transform or predict with copy=False

Fix for bug as described in https://github.com/scikit-learn/scikit-learn/issues/4122#issuecomment-78056759

amueller commented:
Interesting overlap with #4420. I think we should merge this one first and I'll rebase.
Can you add a test that'll make sure transform works with copy=False?

amueller commented:
Hm.... I'm sorry but I think it might be easier if I just fix it in #4420. I didn't have this issue on my mind when I did the PR. The line that you changed is actually not needed at all, as input validation is done beforehand.
At least you didn't seem to have any git trouble.

Commit:
FIX 2 _PLS transform or predict with copy=False

Extended fix for bug as described in https://github.com/scikit-learn/scikit-learn/issues/4122#issuecomment-78056759
This additional commit covers a bug when `Y` is passed with `copy=True`.

hzpc-joostk commented:
Hi, I've add another commit which covers the bug when also `Y` is passed with `copy=True`. I forgot changing that line in the first commit. :) Should I test it with arbitrary training data and try to transform it? I could put both in here then?

amueller commented:
Can you see if you still have issues on master? #4420 got merged, and I hope it fixed all the issues.

hzpc-joostk commented:
Hi @amueller ! After a few days working with the last update (v0.16.1), everything seems to work fine for my purposes. Thanks and cheers.

amueller commented:
great :)
